rest/oidc_api_test.go

```diff
@@ -1,139 +1,816 @@
 package rest
 
 import (
-	"io"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
 	"net/http"
 	"net/http/httptest"
 	"net/url"
+	"regexp"
 	"strings"
 	"testing"
+	"time"
 
 	"github.com/couchbase/sync_gateway/auth"
+	"github.com/couchbase/sync_gateway/base"
+	"github.com/gorilla/mux"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+	"gopkg.in/square/go-jose.v2/jwt"
 )
 
+// A wantErrKind represents the specific error expected.
+type wantError struct {
```

Comment:
I think 'forceError' better describes the usage than 'wantError' - this is being used when you want to force an error of type 'wantErrKind' (which would be better as 'forceErrorType')

To make it clear that 'code' and 'message' are what we expect Sync Gateway to return when we force the error (and not necessarily the code/message for the error itself), I think expectedErrorCode and expectedErrorMessage would be more descriptive.  With those names (type/expectedErrorCode/expectedErrorMessage).